File: C++/Lesson 2 - Arrays/CyclicRotation.cpp
// you can use includes, for example:
#include <vector>
#include <algorithm>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

vector<int> solution(vector<int> &A, int K) {
    // Implement your solution here
    if (A.empty()) {
        return A;
    }
    if (K > A.size()) {
        K %= A.size();
    }
    std::rotate(A.rbegin(), A.rbegin() + K, A.rend());
    return A;
}

File: Python/Lesson 2 - Arrays/CyclicRotation.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A, K):
    # Implement your solution here
    if len(A) == K or len(A) == 1 or K == 0 or len(A) == 0:
        return A
    if K > len(A):
        counter = K // len(A)
        K -= counter * len(A)  
    
    array_front = A[:len(A) - (K)]
    array_back = A[len(A) - (K):]
    array_back.extend(array_front)
    return array_back